casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala

```diff
@@ -95,8 +96,20 @@ class MultiParentCasperImpl[F[_]: Sync: Concurrent: Log: Time: SafetyOracle: Las
 
     import cats.instances.either._
     for {
+      // Save block in block store before waiting for blockProcessingLock
+      // This is called inside block message handler as well, but required here in case block is not received from peer
+      // but proposed by node itself.
+      _ <- BlockStore[F].contains(b.blockHash).ifM(().pure[F], BlockStore[F].put(b))
+      _ <- Log[F].debug(
+            s"Block ${PrettyPrinter.buildString(b.blockHash)} asks for blockProcessingLock."
+          )
```

Comment:
Maybe this log would be useful not only in debug?

Reply:
https://github.com/rchain/rchain/pull/2902/commits/80c745cecfaa8f19ec8e29c0e62452a4e656bcad